tests/test_redis_lock.py

```diff
@@ -188,62 +212,47 @@ def test_plain(conn):
-    released the lock. If there's is overlap (eg: pid1.start < pid2.start < pid1.end) then we
-    got a very bad regression on our hands ...
-    
-    The subprocess being run (check helper.py) will fork bunch of processes and will try to 
-    syncronize them (using the builting sched) to try to acquire the lock at the same time.
     """
-    with TestProcess(sys.executable, HELPER, 'test_no_overlap') as proc:
-        with dump_on_error(proc.read):
-            name = 'lock:foobar'
-            wait_for_strings(proc.read, TIMEOUT, 'Getting %r ...' % name)
-            wait_for_strings(proc.read, TIMEOUT, 'Got lock for %r.' % name)
-            wait_for_strings(proc.read, TIMEOUT, 'Releasing %r.' % name)
-            wait_for_strings(proc.read, TIMEOUT, 'UNLOCK_SCRIPT not cached.')
-            wait_for_strings(proc.read, 10*TIMEOUT, 'DIED.')
-
-            class Event(object):
-                pid = start = end = '?'
-
-                def __str__(self):
-                    return "Event(%s; %r => %r)" % (self.pid, self.start, self.end)
-
-            events = defaultdict(Event)
-            for line in proc.read().splitlines():
-                try:
-                    pid, time, junk = line.split(' ', 2)
-                    pid = int(pid)
-                except ValueError:
-                    continue
-                if 'Got lock for' in junk:
-                    events[pid].pid = pid
-                    events[pid].start = time
-                if 'Releasing' in junk:
-                    events[pid].pid = pid
-                    events[pid].end = time
-            assert len(events) == 125
-
-            # not very smart but we don't have millions of events so it's 
-            # ok - compare all the events with all the other events:
-            for event in events.values():
-                for other in events.values():
-                    if other is not event:
-                        try:
-                            if other.start < event.start < other.end or \
-                               other.start < event.end < other.end:
-                                pytest.fail('%s overlaps %s' % (event, other))
-                        except:
-                            print("[%s/%s]" % (event, other))
-                            raise
+    cond  = multiprocessing.Condition()
+    event = multiprocessing.Event()
+    count = multiprocessing.Value('H', 0)
+
+    def worker_fn(cond, event, count):
+        cond.acquire()
+        lock = Lock(make_conn(), 'lock')
+        event.set()
+        cond.wait()
+
+        lock.acquire()
+
+        cond.acquire()
+        count.value += 1
+        cond.release()
+
+    workers = []
+    for _ in range(125):
+        worker = make_process(target=worker_fn, args=(cond, event, count))
+        worker.start()
+        event.wait(5)
+        event.clear()
+        workers.append(worker)
+
+    time.sleep(0.5)
+    cond.acquire()
+    cond.notify_all()
```

Comment:
Wouldn't this kinda "serialize somewhat" the .acquires? The goal of the test is to simulate a race condition. Something with sched would work better.

Eg, if you add a `sched.enterabs()` inside the worker_fn it would simulate the race condition more accurately. You only need to have a way to send the timestamp to the workers via multiprocessing ... (you pick a timestamp in the future, right after all the workers have been started).